From a69c50fc2b8c35814e195eacb01f0d60feb54dd1 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Sat, 14 Dec 2019 19:26:32 +0100
Subject: [PATCH] src/libtpm2-totp.c: include <endian.h>
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix the following build failure with musl:

src/libtpm2-totp.c: In function ‘tpm2totp_calculate’:
src/libtpm2-totp.c:826:11: warning: implicit declaration of function ‘htobe64’ [-Wimplicit-function-declaration]
     tmp = htobe64(tmp);
           ^~~~~~~

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Retrieved from:
https://github.com/tpm2-software/tpm2-totp/commit/a69c50fc2b8c35814e195eacb01f0d60feb54dd1]
---
 src/libtpm2-totp.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libtpm2-totp.c b/src/libtpm2-totp.c
index a364b7a..2d3f00d 100644
--- a/src/libtpm2-totp.c
+++ b/src/libtpm2-totp.c
@@ -8,6 +8,7 @@
 
 #include <tpm2-totp.h>
 
+#include <endian.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
